catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -90,6 +92,8 @@ public QueryResultIterable(CatalogFramework catalog, Function<Integer, QueryRequ
     class ResultQueryIterator implements Iterator<Result> {
```

Comment:
Agree, we can't make it `static` as it is. We should be able to determine what this `Iterator` really needs and pass that in to its constructor instead of sharing the member variables between the two classes.